packages/augur-ui/src/modules/app/components/main-error-boundary.tsx

```diff
@@ -1,27 +1,31 @@
-import React, { Component } from "react";
-import { Link } from "react-router-dom";
-import makePath from "modules/routes/helpers/make-path";
-import { DEFAULT_VIEW } from "modules/routes/constants/views";
-import { SecondaryButton } from "modules/common/buttons";
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import makePath from 'modules/routes/helpers/make-path';
+import { DEFAULT_VIEW } from 'modules/routes/constants/views';
+import { SecondaryButton } from 'modules/common/buttons';
 
-import Styles from "modules/app/components/main-error-boundary.styles";
-import ButtonStyles from "modules/common/buttons.styles";
+import Styles from 'modules/app/components/main-error-boundary.styles';
+import ButtonStyles from 'modules/common/buttons.styles';
 
 interface MEBState {
   hasError: boolean;
 }
 
 export default class MainErrorBoundary extends Component<{}, MEBState> {
   state: MEBState = {
-    hasError: false
+    hasError: false,
   };
 
-  UNSAFE_componentWillReceiveProps() {
-    if (this.state.hasError) {
-      this.setState({ hasError: false });
+
+  componentDidUpdate(prevProps: {}, prevState: MEBState) {
+    const { hasError } = this.state
+    if (prevState.hasError && hasError) {
+      setTimeout(() => {
```

Comment:
what is the advantage of using `setTimeout` here with 0?

Reply:
@bthaile -- I'm assuming because setting state within componentDidUpdate is a no-no as it creates an update cycle. This looks like a bit of a hack to set the error state...

perhaps I'm wrong?

Reply:
@pgebheim you can setState in componentDidUpdate as long as it's in a conditional, like the if statement it's in.

Reply:
Oh yes -- makes sense!

Reply:
There is a rule for calling setState inside componentDidUpdate so I wrapped it with setTimeout (or setImmediate). Should I remove it?

Reply:
There is a rule for calling setState inside componentDidUpdate so I wrapped it with setTimeout (or setImmediate). Should I remove it?

Reply:
We can safely remove this:
https://reactjs.org/docs/react-component.html#componentdidupdate